File: openharmony-sig/ohos_react_native/tester/harmony/react_native_openharmony/src/main/cpp/RNOHCorePackage/TurboModules/DevMenuTurboModule.cpp
/**
 * Copyright (c) 2024 Huawei Technologies Co., Ltd.
 *
 * This source code is licensed under the MIT license found in the
 * LICENSE-MIT file in the root directory of this source tree.
 */

#include "RNOHCorePackage/TurboModules/DevMenuTurboModule.h"

using namespace facebook;

namespace rnoh {
DevMenuTurboModule::DevMenuTurboModule(
    const ArkTSTurboModule::Context ctx,
    const std::string name)
    : ArkTSTurboModule(ctx, name) {
  methodMap_ = {
      ARK_METHOD_METADATA(show, 0),
      ARK_METHOD_METADATA(reload, 1),
      ARK_METHOD_METADATA(debugRemotely, 1),
      ARK_METHOD_METADATA(setProfilingEnabled, 1),
      ARK_METHOD_METADATA(setHotLoadingEnabled, 1),
  };
}
} // namespace rnoh

File: openharmony-sig/ohos_react_native/tester/harmony/react_native_openharmony/src/main/ets/RNOHCorePackage/turboModules/DevMenuTurboModule.ts
/**
 * Copyright (c) 2024 Huawei Technologies Co., Ltd.
 *
 * This source code is licensed under the MIT license found in the
 * LICENSE-MIT file in the root directory of this source tree.
 */

import type { TurboModuleContext } from '../../RNOH/TurboModule';
import { TurboModule } from '../../RNOH/TurboModule';
import { RNOHLogger } from '../../RNOH/RNOHLogger';

export class DevMenuTurboModule extends TurboModule {
  public static readonly NAME = 'DevMenu';

  private logger: RNOHLogger;

  constructor(protected ctx: TurboModuleContext) {
    super(ctx);
    this.logger = this.ctx.logger.clone("DevMenuTurboModule")
  }

  public show() {
    this.ctx.devMenu.show()
  }

  public reload() {
    this.ctx.devToolsController.reload(undefined)
  }

  public debugRemotely(enableDebug: boolean) {
    this.logger.warn("DevMenu::enableDebug is not supported");
  }

  public setProfilingEnabled(enabled: boolean) {
    this.logger.warn("DevMenu::setProfilingEnabled is not supported");
  }

  public setHotLoadingEnabled(enabled: boolean) {
    this.logger.warn("DevMenu::setHotLoadingEnabled is not supported");
  }
}